internal/characters/childe/abil.go

```diff
@@ -0,0 +1,513 @@
+		mult = meleeBurst[c.TalentLvlBurst()]
+	}
+
+	c.AddTask(func() {
+		d := c.Snapshot(
+			skillName,
+			core.AttackTagElementalBurst,
+			core.ICDTagNone,
+			core.ICDGroupDefault,
+			core.StrikeTypeDefault,
+			core.Hydro,
+			50,
+			mult,
+		)
+		d.Targets = core.TargetAll
+		c.Core.Combat.ApplyDamage(&d)
+	}, "childe-burst-clear", f-5) //random 5 frame
+
+	c.Energy = 0
+	if c.Core.Status.Duration("childemelee") == 0 {
+		c.AddEnergy(20)
+		c.Core.Log.Debugw("Childe ranged burst restoring 20 energy", "frame", c.Core.F, "event", core.LogEnergyEvent, "new energy", c.Energy)
+	} else {
+		// C6 AnnihilationWhen Havoc: Obliteration is cast in Melee Stance, the CD of Foul Legacy: Raging Tide is reset.
+		// This effect will only take place once Tartaglia returns to his Ranged Stance.
+		if c.Base.Cons >= 6 {
+			c.c6 = true
+		}
+	}
+	c.SetCD(core.ActionBurst, 900)
+	return f, a
+}
+
+func (c *char) rtHook() {
+	c.Core.Events.Subscribe(core.OnDamage, func(args ...interface{}) bool {
+		ds := args[1].(*core.Snapshot)
+		t := args[0].(core.Target)
+		crit := args[3].(bool)
+		//childe is active
+		if ds.ActorIndex != c.CharIndex() {
+			return false
+		}
+		//source is hydro
+		if ds.Element != core.Hydro {
+			return false
+		}
+		// dont proc if src from riptides
+		if ds.Abil == "Riptide Flash" || ds.Abil == "Riptide Slash" ||
+			ds.Abil == "Riptide Blast" || ds.Abil == "Riptide Burst" ||
+			ds.Abil == "C4 Riptide Flash" || ds.Abil == "C4 Riptide Slash" {
+			return false
+		}
+
+		switch ds.AttackTag {
+		case core.AttackTagNormal:
+			if c.Core.Status.Duration("childemelee") > 0 {
+				// melee normal
+				if c.rtExpiry[t.Index()] > c.Core.F {
+					// c.Core.Log.Debugw("Riptide Slash checking for tick", "frame", c.Core.F, "event", core.LogCharacterEvent, "dur",
+					// 	c.Core.Status.Duration("childemelee"), "target", t, "fl", c.rtflashICD[t], "sl", c.rtslashICD[t], "rtExpiry", c.rtExpiry[t])
+					if c.rtslashICD[t.Index()] < c.Core.F {
+						c.AddTask(func() {
+							d := c.Snapshot(
+								"Riptide Slash",
+								core.AttackTagElementalArt,
+								core.ICDTagNone,
+								core.ICDGroupDefault,
+								core.StrikeTypeDefault,
+								core.Hydro,
+								25,
+								rtSlash[c.TalentLvlSkill()],
+							)
+							d.Targets = core.TargetAll
+
+							c.Core.Combat.ApplyDamage(&d)
+							c.Core.Log.Debugw("Riptide Slash ticked", "frame", c.Core.F, "event", core.LogCharacterEvent, "dur",
+								c.Core.Status.Duration("childemelee"), "target", t.Index(), "fl", c.rtflashICD[t.Index()], "sl", c.rtslashICD[t.Index()], "rtExpiry", c.rtExpiry[t.Index()])
+						}, "Riptide Slash", 5)
+						c.rtslashICD[t.Index()] = c.Core.F + 90 //1.5s icd
+					}
+				}
+
+				// A4:Sword of TorrentsWhen Tartaglia is in Foul Legacy: Raging Tide’s Melee Stance,
+				// on dealing a CRIT hit, Normal and Charged Attacks apply the Riptide status effect to opponents.
+				if crit {
+					c.applyRT(t.Index())
+				}
+			}
+		case core.AttackTagExtra:
```

Comment:
Maybe I'm missing something, but isn't this mostly just a repeat of the NormalAttack case? Can't we just have the above case match two values and mostly drop this? After moving the aim mode stuff up?